debugger/src/debugger.rs

```diff
@@ -1072,18 +1076,158 @@ impl Debugger {
+                                            id,
+                                        )
+                                        .ok();
+                                }
+                                probe_rs::flashing::ProgressEvent::FailedFilling => {
+                                    debug_adapter
+                                        .update_progress(1.0, Some("Reading Old Pages Failed!"), id)
+                                        .ok();
+                                }
+                                probe_rs::flashing::ProgressEvent::FinishedFilling => {
+                                    debug_adapter
+                                        .update_progress(
+                                            1.0,
+                                            Some("Reading Old Pages Complete!"),
+                                            id,
+                                        )
+                                        .ok();
+                                }
+                                probe_rs::flashing::ProgressEvent::StartedErasing => {
+                                    debug_adapter
+                                        .update_progress(0.0, Some("Erasing Sectors ..."), id)
+                                        .ok();
+                                }
+                                probe_rs::flashing::ProgressEvent::SectorErased {
+                                    size, ..
+                                } => {
+                                    flash_progress.sector_size_done += size as usize;
+                                    debug_adapter
+                                        .update_progress(
+                                            flash_progress.sector_size_done as f64
+                                                / flash_progress.total_sector_size as f64,
+                                            Some("Erasing Sectors ..."),
+                                            id,
+                                        )
+                                        .ok();
+                                }
+                                probe_rs::flashing::ProgressEvent::FailedErasing => {
+                                    debug_adapter
+                                        .update_progress(1.0, Some("Erasing Sectors Failed!"), id)
+                                        .ok();
+                                }
+                                probe_rs::flashing::ProgressEvent::FinishedErasing => {
+                                    debug_adapter
+                                        .update_progress(1.0, Some("Erasing Sectors Complete!"), id)
+                                        .ok();
+                                }
+                                probe_rs::flashing::ProgressEvent::StartedProgramming => {
+                                    debug_adapter
+                                        .update_progress(0.0, Some("Programming Pages ..."), id)
```

Comment:
I think it would be helpful to add the % complete into the message text. That way a user can see the progress in the VSCode status bar, without having to click into to the associated notifications.

Reply:
Hmm it should give a popup on the bottom right with the progress? Does this not happen for you?

Reply:
The popup only happens when I click on the status bar where the flashing progress shows the text message. I'd prefer to NOT have the popup, and just see the status bar unobtrusively keep track of the percentage complete.

Reply:
Hmm I like the popup for this case since you expect something to happen and it disappears automatically after it's done I think it's very unobtrusive :) Generally I hate random popups but here I like it :)

Reply:
I understand. You can leave the events as they are, and the user can control in VSCode if they want them to auto popup (you do) or not (i do). What I'm suggesting is to just add the % completion into the current text of the message parameter of the event body.  Then we get the best of both worlds ;-)

Reply:
Ah, yes, I was not argueing against adding the % in the text :) Sorry if that was unclear :/

Reply:
I just tried the change ... exactly what I wished for. Thank :)